app/services/msg/depart.rb

```diff
@@ -0,0 +1,28 @@
+class Msg::Depart < Msg::Base
+  # @returns String
+  def respond
+    # this block potentially involves several write operations, which we want 
+    # to be atomic.
+    Timecard.transaction do
+      incomplete_timecards = @person.timecards.where(status: "Incomplete")
+      
+      # If the number of incomplete timecards is not exactly 1, create an error 
+      # entry so that they can be manually reconciled
+      timecards_count = @person.timecards.where(status: "Incomplete").count
+      if timecards_count != 1
+        # set all incomplete timecards to error state
+        incomplete_timecards.update_all(status: "Error")
+        # record the depart time as a new timecard entry
+        Timecard.create!(person_id: @person.id,
+                         status: "Error",
+                         end_time: Time.now)
+        return "Error: #{timecards_count} incomplete timecards."
+      end
+      
+      timecard = incomplete_timecards.first
```

Comment:
On my first scan, I missed that we would not fall through to line 22, due to line 19. What do you think of doing the happy path ( timecards_count =1 ) as an if block, and then an else block for not equaling one ? Lines 13-19 would be in the else block. I realize your code has the same effect, but I think that would better illuminate the two paths.

Your thoughts ?
Thanks for your help. I'll be doing a deployment before Wed evening, so having this is great !
Kevin